core/src/main/java/com/yahoo/oak/Slice.java

```diff
@@ -251,13 +236,19 @@ public int getLength() {
 
     @Override
     public long getAddress() {
-        return ((DirectBuffer) buffer).address() + getOffset();
+        return memAddress + getOffset();
     }
 
     @Override
     public String toString() {
         return String.format("Slice(blockID=%d, offset=%,d, length=%,d, version=%d)", blockID, offset, length, version);
     }
+    
+    //must never be called internally kept in here to beautify the interface
+    @Override
+    public ByteBuffer getByteBuffer() { 
+        throw new RuntimeException(); 
+    }
```

Comment:
Please remove this method entirely. 
Note that `Slice` should not implement `OakUnsafeDirectBuffer` anymore, so please remove `implements OakUnsafeDirectBuffer` from `Slice` declaration.
Also, please remove `order()` from `OakBuffer` API. We use native order now.

Reply:
I think that Slice should still implement the methods of OakUnsafeDirectBuffer, we use these methods in ScopedReadBuffer for example.
what can I do is to remove the implements from the OakUnsafeDirectBuffer but still have the method, is that what you have meant?

Reply:
Yes. Leave the methods you need and remove the ones that are redundant (e.g., `getByteBuffer()`).
Anyway, it should not be declared with `implements OakUnsafeDirectBuffer`.